cpp/Json.hpp

```diff
@@ -350,6 +368,49 @@ namespace hsd {
         }
     };
 
+    template <typename CharT>
+    basic_string_view<CharT> JsonValue::as_str() { 
+        return try_as<JsonString<CharT>>(JsonValueType::String).unwrap().value(); 
+    }
+
+    i32 JsonValue::as_num() { 
+        return try_as<JsonNumber>(JsonValueType::Number).unwrap().value(); 
+    }
+
+    bool JsonValue::as_bool() { 
+        if (type() == JsonValueType::True)
+        {
+            return true;
+        }
+        else if (type() == JsonValueType::False)
+        {
+            return false;
+        }
+
+        // throw
+        Result<bool, runtime_error> res = runtime_error{"Cast to wrong type"};
+        return res.unwrap();
+    }
+
+    template <typename CharT>
+    JsonValue& JsonValue::access(const basic_string_view<CharT>& key)
+    {
+        auto& _map = try_as<JsonObject<CharT>>(JsonValueType::Object).unwrap().values();
+        return *_map.at(key).unwrap();
+    }
+
+    template <typename CharT, usize N>
+    JsonValue& JsonValue::access(const CharT (&key)[N])
```

Comment:
I'm pretty sure we should rather move this into `basic_string_view`, rather than do it manually here

Reply:
it's moved into the string view and i've added literals